e2e/singleserver_singleagent_test.go

```diff
@@ -16,10 +17,11 @@ import (
 )
 
 func TestSingleServer_SingleAgent_StaticCount(t *testing.T) {
-	serviceHost := "konnectivity-server.kube-system.svc.cluster.local"
+	serverServiceHost := "konnectivity-server.kube-system.svc.cluster.local"
+	agentServiceHost := "konnectivity-agent.kube-system.svc.cluster.local"
 	adminPort := 8093
 
-	serverDeploymentCfg := DeploymentConfig{
+	serverStatefulSetCfg := StatefulSetConfig{
```

Comment:
why change from deployment to statefulset ?

Reply:
StatefulSets allow for easier addressing of specific pods that belong to a service.

Reply:
With Imran here. Agent should not need to maintain state across instances. So from a Kubernetes perspective it should be stateless and hence should run in a deployment not a statefulset. Worried that running it in statefulset in a test could mask a problem.

Reply:
Hmm, do you know of a way to get metrics from a specific pod? That's the reason I was using a StatefulSet.